### PR TITLE
Add a pivot for GameObject

### LINKED ISSUE #112: Add Pivot to GameObject
Adding a pivot option to `Util::GameObject` could allow users to offset their `m_Drawable` objects. e.g. users could rotate an image around the pivot, some users might find it convenient to anchor the text to the left

### PR DESCRIPTION
**Pull request by wj461**
- resolve #112 

### Changes
- Change `Drawable` Interface
```diff
- Draw(const Util::Transform &transform, const float zIndex)
+ Draw(const Core::Matrices &data)
```

- Add `pivot` in `GameObject`

### CODE REVIEW

File: src/Util/GameObject.cpp
Comment (ntut-rick) [inline src/Util/GameObject.cpp:1]: I'm not sure what this code can do. It seems like `m_Pivot` and `m_Transform.translation` serve the same function.

File: src/Util/GameObject.cpp [hunk 1]
```diff
@@ -7,7 +8,13 @@ void GameObject::Draw() {
...
     }
 
-    m_Drawable->Draw(m_Transform, m_ZIndex);
+    Util::Transform offsetTransform {
+        m_Transform.translation - m_Pivot,
+        m_Transform.rotation,
+        m_Transform.scale,
+    };
+
+    m_Drawable->Draw(offsetTransform, m_ZIndex);
```
Comment (ntut-rick) [inline src/Util/GameObject.cpp:17]: I'm not sure what this code can do. It seems like `m_Pivot` and `m_Transform.translation` serve the same function.

**Review by Dragonfly911117 — APPROVED** [review]
WOMM

**Comment by ntut-rick** [comment]
I think this feature is not worthy to breaking the internal implement.
IMO here's a better way, but we are still discussing:
```diff
- void Draw(const Util::Transform &transform, const float zIndex,
-           const glm::vec2 &pivot) override;
+ void Draw(const glm::mat4 &mat4);
```

**Review by ntut-rick — CHANGES_REQUESTED** [review]
I think this feature is not worthy to breaking the internal implement.
IMO here's a better way, but we are still discussing:
```diff
- void Draw(const Util::Transform &transform, const float zIndex,
-           const glm::vec2 &pivot) override;
+ void Draw(const glm::mat4 &mat4);
```

**Review by NOOBDY — APPROVED** [review]

**Review by ntut-rick — APPROVED** [review]
agreeGe

**Pull request merged**